### PR TITLE
[dem] Histogram with the number of neighbours is now optional

### LINKED ISSUE #7019: DEM always prints number_of_neighbours_histogram.txt
can this be made optional? or is it already possible to disable this?
Thx

(discovered in CoSim tests)

### PR DESCRIPTION
**Pull request by maceligueta**
Closes #7019
A new variable in the json was added. A new 'if' in the AnalysisStage has been added.
We are completely aware that the flat structure in the ProjectParametersDEM.json is outdated and inconvenient, but re-arranging it is delicate and will require a bit of time.

### CODE REVIEW

File: applications/DEMApplication/python_scripts/dem_default_input_parameters.py [hunk 1]
```diff
@@ -172,6 +172,7 @@ def GetDefaultInputParameters():
             "PostFaceNormalImpactVelocity"     : false,
             "PostFaceTangentialImpactVelocity" : false,
             "PostControlModule"                : false,
+            "PostNumberOfNeighboursHistogramOption" : false,
```
Comment (philbucher) [inline applications/DEMApplication/python_scripts/dem_default_input_parameters.py]:
```suggestion
            "output_configuration" : {
                "print_number_of_neighbours_histogram" : false
            }
```

**Review by philbucher — DISMISSED** [review]
thx for the fix

I have two suggestions:
Either what I suggested, or making a process out of it and (if this output is wanted) adding it to the list of (output-) processes. Last one would be consistent with the other bigger apps (Fluid & Structure etc)

**Review by philbucher — APPROVED** [review]

**Comment by maceligueta** [comment]
I am a bit lost. The Windows CI can't compile. But I just changed two lines of Python...

**Comment by philbucher** [comment]
See #7067

**Comment by maceligueta** [comment]
Thanks for merging master, @philbucher !

**Pull request merged**